unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2100,35 +2107,9 @@ protected SimObjPhysics[] GetAllVisibleSimObjPhysics(Camera agentCamera, float m
                         sop = item.GetComponentInParent<SimObjPhysics>();
 
                         //now we have a reference to our sim object 
-                        if (sop) 
+                        if (sop && isSimObjVisible(agentCamera, sop, true))  
                         {
-                            //check against all visibility points, accumulate count. If at least one point is visible, set object to visible
-                            if (sop.VisibilityPoints.Length > 0) 
-                            {
-                                Transform[] visPoints = sop.VisibilityPoints;
-                                int visPointCount = 0;
-
-                                foreach (Transform point in visPoints) 
-                                {
-                                    //if this particular point is in view...
-                                    if (CheckIfVisibilityPointInViewport(sop, point, agentCamera, true)) 
-                                    {
-                                        visPointCount++;
-                                    }
-                                }
-
-                                //if we see at least one vis point, the object is "visible"
-                                if (visPointCount > 0) 
-                                {
-                                    #if UNITY_EDITOR
```

Comment:
I would add this editor define back so that debug highlighting of objects still works while in editor mode.

Reply:
actually wait, I believe this is fine because this was all effectively moved to the isSimObjVisible call